### PR TITLE
[WIP] Ng2 tracker service

### PR DESCRIPTION
**Pull request by flaviencathala**
@christopherthielen Could you please take a look we still have $rootScope in tracker service.

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/shared/tracker/_tracker.service.ts [hunk 1]
```diff
@@ -39,7 +37,7 @@ export class <%=jhiPrefixCapitalized%>TrackerService {
     connect () {
         if (this.connectedPromise === null) this.connection = this.createConnection();
         //building absolute path so that websocket doesnt fail when deploying with a context path
-        var loc = this.$window.location;
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/tracker/_tracker.service.ts]: we should either continue to use the`Window` object from ng2 or do something like http://juristr.com/blog/2016/09/ng2-get-window-ref/

**Comment by dimeros** [comment]
I see that for some reason, the connection cannot be established. 
Debugging it says that the sockJs is undefined. @flaviencathala is it reproducible to your side?

**Comment by flaviencathala** [comment]
@dimeros Nope I've never seen this error

**Comment by deepu105** [comment]
@flaviencathala are you doing anything here?

File: generators/client/templates/angular/src/main/webapp/app/shared/tracker/_tracker.service.ts (hunk 1 above)
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/tracker/_tracker.service.ts]: please change this

**Comment by deepu105** [comment]
@flaviencathala I'm closing this. If you are working on this please open a new PR with cleaner commits

**Pull request closed without merging**